File: Services/entities/Transaction.ts
import { Column, Entity, ManyToOne, PrimaryGeneratedColumn } from 'typeorm';
import { User } from './User';


@Entity('transactions')
export class Transaction{

    @PrimaryGeneratedColumn('uuid')
    id?: string


    @Column('varchar')
    baseCurrency!: string

    @Column('varchar')
    changeCurrency!: string

    @Column('double')
    transactionDate?: number

    @Column('double')
    amount = 0

    @Column('double')
    rate = 0

    @Column('varchar')
    status!:  'accepted' | 'processing' | 'finished' | 'rejected'

    //relations
    @ManyToOne(() => User, u => u.transactions)
    transactionActor?: User

}

File: Apps/SampleWebApp/AdaptedApp/ExMoney/Models/Transaction.cs
namespace ExMoney.Models;


public class Transaction
{
    public string Id { get; set; }

    public string BasCurrency { get; set; }

    public string ChangeCurrency { get; set; }

    public DateTime TransactionDate { get; set; }

    public double Amount { get; set; }

    public double Rate { get; set; }
    
    // value inferred from TransactionStatus enum    
    public string Status { get; set; }

}


public enum TransactionStatus
{
    Accepted,
    Processing,
    Finished,
    Rejected
}
